runelite-client/src/main/java/net/runelite/client/plugins/worldmap/WorldMapPlugin.java

```diff
@@ -271,4 +287,57 @@ private void updateShownIcons()
 			}).map(TeleportPoint::new)
 			.forEach(worldMapPointManager::add);
 	}
+
+	private void updateQuestStartPointIcons()
+	{
+		if (!config.questStartTooltips()) return;
```

Comment:
This causes this to not remove the quests from the map when quest start tooltips are toggled off after already showing. That is why it was removing the points before checking the new config. I also don't see a need to move it out of the `updateShowIcons` call.

Reply:
Got it- can move the point removing back to `updateShowIcons`. However, the rest didn't work when it was in `updateShowIcons`, I was getting an error when accessing the `client` for quest state. So I had to move it to `onWidgetLoaded`